controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -182,6 +165,33 @@ func (w *Workload) updateManagedEtcdConditions(ctx context.Context, controlPlane
 	})
 }
 
+func (w *Workload) getCurrentEtcdMembers(ctx context.Context, machine *clusterv1.Machine, nodeName string) ([]*etcd.Member, error) {
+	// Create the etcd Client for the etcd Pod scheduled on the Node
+	etcdClient, err := w.etcdClientGenerator.forFirstAvailableNode(ctx, []string{nodeName})
+	if err != nil {
+		conditions.MarkUnknown(machine, controlplanev1.MachineEtcdMemberHealthyCondition, controlplanev1.EtcdMemberInspectionFailedReason, "Failed to connect to the etcd pod on the %s node: %s", nodeName, err)
+		return nil, errors.Wrapf(err, "failed to get current etcd members: failed to connect to the etcd pod on the %s node", nodeName)
```

Comment:
I know we're ignoring those error messages one layer above, but for me that was the least awkward way to return the info that retrieving current members failed. Just returning a nil slice would be to brittle in my opinion.